### PR TITLE
wasi-sdk-pthread.cmake: add --import-memory

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

**Comment by TerrorJack** [comment]
Shall we also add `--shared-memory` and a large `--max-memory` default value?

**Comment by yamt** [comment]
> Shall we also add `--shared-memory` and a large `--max-memory` default value?

`--shared-memory` is automatically added for -pthead.

i don't think there is a good single value for `--max-memory`. it depends on apps.

**Comment by yamt** [comment]
i marked this a draft because this needs llvm >=16.

**Comment by abrown** [comment]
I am also wondering if we should rename this file `wasi-sdk-threads.cmake`?

**Comment by yamt** [comment]
> I am also wondering if we should rename this file `wasi-sdk-threads.cmake`?

i don't think so as what this provides is pthread.

**Comment by yamt** [comment]
this is ready as we now ship llvm 16.

File: wasi-sdk-pthread.cmake [hunk 1]
```diff
@@ -9,6 +9,11 @@ set(CMAKE_SYSTEM_PROCESSOR wasm32)
 set(triple wasm32-wasi-threads)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+# wasi-threads requries --import-memory.
```
Comment (abrown) [inline wasi-sdk-pthread.cmake:12]:
```suggestion
# wasi-threads requires --import-memory.
```

**Review by abrown — APPROVED** [review]

File: wasi-sdk-pthread.cmake (hunk 1 above)
Comment (yamt) [reply wasi-sdk-pthread.cmake]: fixed. thank you.

**Pull request merged**